tools/daemon/flow.c

```diff
@@ -611,6 +583,102 @@ int del_flow(struct store_pid *pid_value, struct store_flow *value)
 	return rc;
 }
 
+static int add_flow_egress(struct store_pid *pid_value, struct store_flow *value)
+{
+	int rc = 0;
+	pid_t pid = pid_value->pid;
+	struct list_head *cur_entry = NULL;
+	struct store_flow *cur_flow = NULL;
+	char if_name[IF_NAMESIZE];
+	char tap_name[IF_NAMESIZE];
+	char *out_buf = NULL;
+
+	errno = 0;
+	if (NULL == if_indextoname(value->if_id, if_name)) {
+		log_error("[%d] network interface is not found by index %d errno %d (%s)\n",
+				pid, value->if_id, errno, strerror(errno));
+		rc = -errno;
+		goto err;
+	}
+
+	if (NULL == if_indextoname(value->tap_id, tap_name)) {
+		log_error("[%d] tap interface is not found by index %d errno %d (%s)\n",
+				pid, value->tap_id, errno, strerror(errno));
+		rc = -errno;
+		goto err;
+	}
+
+	/* Egress rules should be created for new tap device
+	 */
+	list_for_each(cur_entry, &pid_value->flow_list) {
+		cur_flow = list_entry(cur_entry, struct store_flow, item);
+		if (value->tap_id == cur_flow->tap_id) {
+			break;
+		}
+	}
+	if (cur_entry == &pid_value->flow_list) {
+		struct ifaddrs *ifaddr, *ifa;
+		char *addr = NULL;
+		int handle = 1;
+
+		/* This cleanup is done just to support verification */
+		sys_exec("tc qdisc del dev %s handle ffff: ingress > /dev/null 2>&1 || echo $?", tap_name);
+
+		/* Create rules to process ingress trafic on tap device */
+		out_buf = sys_exec("tc qdisc add dev %s handle ffff: ingress > /dev/null 2>&1 || echo $?", tap_name);
+		if (NULL == out_buf || (out_buf[0] != '\0' && out_buf[0] != '0')) {
+			log_error("[%d] failed tc qdisc add dev %s output: %s\n",
+					pid, tap_name, (out_buf ? out_buf : "n/a"));
+			rc = -EFAULT;
+			goto err;
+		}
+
+		if (!getifaddrs(&ifaddr)) {
```

Comment:
Can we avoid this iteration for each egress flow ? why not passing the interface flags from VMA  ?